runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkRunner.java

```diff
@@ -78,6 +102,33 @@ public SparkPipelineResult run(Pipeline pipeline) {
     result.waitUntilFinish();
     assertThat(result, testPipelineOptions.getOnCreateMatcher());
     assertThat(result, testPipelineOptions.getOnSuccessMatcher());
+    // if the pipeline was executed in streaming mode, validate aggregators.
+    if (isStreaming) {
+      // validate assertion succeeded (at least once).
+      int success = result.getAggregatorValue(PAssert.SUCCESS_COUNTER, Integer.class);
+      assertThat("Success aggregator should be greater than zero.", success, not(0));
+      // validate assertion didn't fail.
+      int failure = result.getAggregatorValue(PAssert.FAILURE_COUNTER, Integer.class);
+      assertThat("Failure aggregator should be zero.", failure, is(0));
+    }
     return result;
   }
+
+  private static class AdaptedBoundedAsUnbounded extends PTransform<PBegin, PCollection<?>> {
```

Comment:
Even though the uses of this are all dynamically typed, adding a type parameter would increase the type safety and readability within this class definition, for example preventing accidentally returning `PCollection<ValueWithRecordId<T>>`. I mean something like this:

```java
class AdaptedBoundedAsUnbounded<T> extends PTransform<PBegin, PCollection<T>> {
  ...
  AdaptedBoundedAsUnbounded(BoundedReadFromUnboundedSource<T> source) { ... }
}
```